### PR TITLE
#2 Support for Custom Ranges

### PR DESCRIPTION
**Pull request by iv4n-ga6l**
Allow users to specify custom ranges for the Int, Float, and Duration functions instead of always starting at 0.

### CODE REVIEW

File: chaos.go [hunk 1]
```diff
@@ -38,6 +56,15 @@ func Float[F constraints.Float](n F, seed ...any) F {
 	return F(r.Float64()) * n
 }
 
+// FloatInRange returns a deterministic float between min and max.
+func FloatInRange(min, max float64, seed ...any) float64 {
```
Comment (raphoester) [inline chaos.go]: Functions that can panic should contain the "Must" indicator

File: chaos.go [hunk 2]
```diff
@@ -26,6 +35,15 @@ func Duration(n time.Duration, seed ...any) time.Duration {
 	return time.Duration(Int(n.Nanoseconds(), seed))
 }
 
+// DurationInRange returns a deterministic duration between min and max.
+func DurationInRange(min, max time.Duration, seed ...any) time.Duration {
```
Comment (raphoester) [inline chaos.go]: Functions that can panic should contain the "Must" indicator

File: chaos.go [hunk 3]
```diff
@@ -16,6 +16,15 @@ func Int[I constraints.Integer](n I, seed ...any) I {
 	return I(r.Int63n(int64(n)))
 }
 
+// IntInRange returns a deterministic integer between min and max (inclusive).
+func IntInRange(min, max int, seed ...any) int {
```
Comment (raphoester) [inline chaos.go]: Functions that can panic should contain the "Must" indicator

**Review by raphoester — CHANGES_REQUESTED** [review]
fix conflicts with latest merge and it's good to go

File: chaos.go (hunk 1 above)
Comment (iv4n-ga6l) [reply chaos.go]: It means include "Must" in the function name ?

**Comment by raphoester** [comment]
what do you think @IvanGael ? I adapted your ideas to the new package organization (with methods instead of plain functions)

**Comment by raphoester** [comment]
Maybe add some tests before merging ?

**Comment by iv4n-ga6l** [comment]
It seems you change a little bit the implementation. The functions Int, MustIntInRange, Bool, Duration, MustDurationInRange, Time, Float, MustFloatInRange, ..Etc have been removed in chaos.go

Should I pull the latest changes from the main branch before adding some tests?

**Comment by raphoester** [comment]
Yea, `chaos.go` is only dedicated to the `Chaos` object now. 
Functions and methods for native types have been moved to `basic_types.go`.
Therefore their tests should be moved to `basic_types_test.go`.

Also the behavior has been adapted to avoid unnecessary redundancy : *Range* functions become *Between*

**Pull request merged**